src/mantle/framework/bootstrap/class-load-configuration.php

```diff
@@ -137,33 +151,94 @@ protected function get_configuration_directories( Application $app ): array {
 	}
 
 	/**
-	 * Load specific configuration files to a repository
+	 * Find the configuration files to load.
 	 *
-	 * @param string[]            $files Files to load.
-	 * @param Repository_Contract $repository Repository to load to.
-	 * @param bool                $merge Flag to merge the configuration instead of overwriting.
+	 * @param Application $app Application instance.
+	 * @return array{global: array<string, string[]>, env: array<string, array<string, string[]>>}
 	 */
-	protected function load_configuration_to_repository( array $files, Repository_Contract $repository, bool $merge = false ) {
-		$filesystem = new Filesystem();
+	protected function get_configuration_files( Application $app ): array {
+		$files = [
+			'global' => [],
+			'env'    => [],
+		];
 
-		foreach ( $files as $key => $root_config_file ) {
-			$config = $filesystem->get_require( $root_config_file );
+		$finder = Finder::create()
+			->files()
+			->name( '*.php' )
+			->depth( '< 2' ) // Only descend two levels.
+			->in( $this->get_configuration_directories( $app ) );
 
-			if ( ! $merge || ! $repository->has( $key ) ) {
-				$repository->set( $key, $config );
-				continue;
+		foreach ( $finder as $file ) {
+			$name = basename( $file->getRealPath(), '.php' );
+
+			$dirname     = basename( dirname( $file->getRealPath() ) );
+			$environment = match ( $dirname ) {
+				// Ignore the 'config' directory as an environment.
+				'config', 'configuration' => null,
```

Comment:
Not really sure, but do any of the changes for loading and merging configuration in this file have performance considerations we should think about?